scripts/engineering/aggregate_phase44d_carry_aware_experimental_family_matrix.py

```
@@ -0,0 +1,364 @@
+
+
+def validate_family_rows(rows: list[dict[str, str]], *, spec: FamilySpec) -> tuple[str, str, str, int]:
+    seen = set()
+    timing_mode = rows[0].get("timing_mode", "").strip()
+    timing_policy = rows[0].get("timing_policy", "").strip()
+    timing_unit = rows[0].get("timing_unit", "").strip()
+    timing_runs = int(rows[0].get("timing_runs", "0"))
+    if timing_mode != "measured_median":
+        raise SystemExit(
+            f"{spec.input_path} must contain measured_median rows; got {timing_mode!r}"
+        )
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)